src/common/atomic.h

```diff
@@ -104,7 +104,7 @@ forceinline volatile int64 InterlockedExchange64(volatile int64 *target, int64 v
 // The __sync functions are available on x86 or ARMv6+
 #if !defined(__x86_64__) && !defined(__i386__) \
 	&& ( !defined(__ARM_ARCH_VERSION__) || __ARM_ARCH_VERSION__ < 6 )
-#error Your Target Platfrom is not supported
+//#error Your Target Platfrom is not supported
```

Comment:
Why keep the check if nothing happens on failure?

Reply:
technically here should be error, but i not sure conditions what to really check.
current condition show error on new arm arches, and this is wrong.

Reply:
Okay, I see.

Reply:
This is because we don't currently detect ARMv8 - see cbasetypes.h:

```
// Standardize the ARM platform version, if available (the only values we're interested in right now are >= ARMv6)
#if defined(__ARMV6__) || defined(__ARM_ARCH_6__) || defined(__ARM_ARCH_6J__) || defined(__ARM_ARCH_6K__) \
	|| defined(__ARM_ARCH_6Z__) || defined(__ARM_ARCH_6ZK__) || defined(__ARM_ARCH_6T2__) // gcc ARMv6
#define __ARM_ARCH_VERSION__ 6
#elif defined(__ARM_ARCH_7A__) || defined(__ARM_ARCH_7R__) || defined(__ARM_ARCH_7S__) // gcc ARMv7
#define __ARM_ARCH_VERSION__ 7
#elif defined(_M_ARM) // MSVC
#define __ARM_ARCH_VERSION__ _M_ARM
#else
#define __ARM_ARCH_VERSION__ 0
#endif
```

In order to detect it, and if https://opensource.apple.com/source/WTF/WTF-7601.4.2/wtf/Platform.h.auto.html is accurate (it probably is, considering Apple's heavy use of ARM processors) we can probably simply add the blocks equivalent to:

```
#elif defined(__ARM_ARCH_8__)
#define WTF_ARM_ARCH_VERSION 8
```

and

```
/* RVCT sets _TARGET_ARCH_ARM */
#elif defined(__TARGET_ARCH_ARM)
#define WTF_ARM_ARCH_VERSION __TARGET_ARCH_ARM
```

(the last one being optional since we don't officially support RVCT, and I have no ARM machines to test things on)